### PR TITLE
COMMERCE-1172 Only get personal accounts if the user isn't deactivated

### PR DESCRIPTION
**Pull request by alecsloan**
@marco-leo Let me know if you know of a better fix. I applied it to the AccountIndexer but this doesn't work with a User being deactivated and requires a manual re-index after each deactivation.

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

File: commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountLocalServiceImpl.java [hunk 1]
```diff
@@ -562,7 +563,14 @@ protected SearchContext buildSearchContext(
 				indexer.delete(companyId, document.getUID());
 			}
 			else if (commerceAccount != null) {
-				commerceAccounts.add(commerceAccount);
+				User user = _userLocalService.getUser(
+					commerceAccount.getUserId());
```
Comment (marco-leo) [inline commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountLocalServiceImpl.java:737]: wrong user Id

**Pull request merged**